pkg/policyfilter/dummy.go

```diff
@@ -16,11 +14,11 @@ type dummy struct {
 }
 
 func (s *dummy) AddPolicy(polID PolicyID, namespace string, podSelector *slimv1.LabelSelector) error {
-	return fmt.Errorf("policyfilter is disabled")
+	return nil
```

Comment:
The problem with not having an error here, is that we rely on this to not accept namespaced policies if policyfilter is disabled. So this, intentionallygi returns an error.

When we add a policy, we call `updatePolicyFilter`
https://github.com/cilium/tetragon/blob/79aa12a01b97ae30315c352712ff1dec12aea047/pkg/sensors/handler.go#L109

Where we want `AddPolicy` to fail if policyfilter is disabled, and reject the policy.
https://github.com/cilium/tetragon/blob/79aa12a01b97ae30315c352712ff1dec12aea047/pkg/sensors/handler.go#L79-L90


Would things work if we only return an error from `AddPolicy` but return `nil` from `DelPolicy`?


It's probably a good idea to have unit tests for this. If you can do it in this PR, that's great! Otherwise, let's create an issue to track this.

Reply:
Took a stab at it in: https://github.com/cilium/tetragon/pull/1200. I used a different fix. PTAL.